src/app/wishlists/wishlist-fetch.ts

```diff
@@ -0,0 +1,76 @@
+import { toWishList } from './wishlist-file';
+import { t } from 'app/i18next-t';
+import _ from 'lodash';
+import { showNotification } from 'app/notifications/notifications';
+import { loadWishLists } from './actions';
+import store from 'app/store/store';
+import { ThunkResult, RootState } from 'app/store/reducers';
+import { ThunkDispatch } from 'redux-thunk';
+import { AnyAction } from 'redux';
+
+export function fetchWishList(showAlert: boolean): ThunkResult<Promise<void>> {
+  return async (dispatch, getState) => {
+    const wishListSource = getState().settings.wishListSource;
+
+    if (!wishListSource) {
+      return;
+    }
+
+    fetch(wishListSource)
+      .then((result) => result.text())
+      .then((resultText) =>
+        transformAndStoreWishList(resultText, 'Fetch Wish List', showAlert, dispatch)
+      );
+  };
+}
+
+export function transformAndStoreWishList(
```

Comment:
Make this another thunk action, which can be called by both the other function and your component. Then you won't have to make dispatch optional.

Reply:
Yeah this makes it way less smelly.